### PR TITLE
WIP: Out of Bag

### PR DESCRIPTION
**Pull request by roshanr10**

### CODE REVIEW

File: src/main/java/frc/team4909/robot/subsystems/climber/commands/BothLiftDown.java [hunk 1]
```diff
@@ -12,7 +13,11 @@ public BothLiftDown(){
     }
 
     public void execute() {
-        Robot.climberSubsystem.setStiltsClimbSpeed(RobotConstants.liftClimbSpeed);
+        SmartDashboard.putString("In Both Lift Down", "7");
```
Comment (roshanr10) [inline src/main/java/frc/team4909/robot/subsystems/climber/commands/BothLiftDown.java]: uhhhhh what is this even intended to say? @ashwinc12

File: src/main/java/frc/team4909/robot/subsystems/climber/commands/BothLiftUp.java [hunk 2]
```diff
@@ -12,7 +13,10 @@ public BothLiftUp(){
     }
 
     public void execute() {
+        SmartDashboard.putString("In both lift up", "11");
```
Comment (roshanr10) [inline src/main/java/frc/team4909/robot/subsystems/climber/commands/BothLiftUp.java]: same as above?

File: src/main/java/frc/team4909/robot/Robot.java [hunk 3]
```diff
@@ -150,8 +150,8 @@ public void robotInit() {
     manipulatorGamepad.buttonHeld(BionicF310.RB, new HatchPanelIntakeOpen());
 
     /* Climber */
-    climberGamepad.buttonHeld(BionicF310.RB, new StiltsUpOnly());
-    climberGamepad.buttonHeld(BionicF310.LB, new StiltsDownOnly());
+    // climberGamepad.buttonHeld(BionicF310.RB, new StiltsUpOnly());
```
Comment (roshanr10) [inline src/main/java/frc/team4909/robot/Robot.java]: why the change? has this implemented elsewhere ?

File: src/main/java/frc/team4909/robot/Robot.java [hunk 4]
```diff
@@ -155,13 +162,30 @@ public void robotInit() {
...
+    // SmartDashboard.putData(new ZeroElevator());
+    // SmartDashboard.putData(new ZeroStilts());
+
+
+
+
 
 
-    SmartDashboard.putData(new ZeroElevator());
-    SmartDashboard.putData(new ZeroStilts());
```
Comment (roshanr10) [inline src/main/java/frc/team4909/robot/Robot.java]: can we leave the button there but hide it so we have it if necessary?

File: src/main/java/frc/team4909/robot/Robot.java [hunk 5]
```diff
@@ -175,9 +199,13 @@ public void robotInit() {
    */
   @Override
   public void robotPeriodic() {
-
+    SmartDashboard.putNumber("Time Remaining", DriverStation.getInstance().getMatchTime());//Useful Method to get match data.
```
Comment (roshanr10) [inline src/main/java/frc/team4909/robot/Robot.java]: this needs to be tested in practice mode

File: src/main/java/frc/team4909/robot/Robot.java [hunk 6]
```diff
@@ -175,9 +199,13 @@ public void robotInit() {
...
     // process();
     Scheduler.getInstance().run();
+
+    SmartDashboard.putNumber("Elevator encoder position", Robot.elevatorSubsystem.getPosition());
```
Comment (roshanr10) [inline src/main/java/frc/team4909/robot/Robot.java]: can we capitalize these variable names? nit picky but it looks nicer if we’re going to use SmartDashboard or Shuffleboard as backup

File: src/main/java/frc/team4909/robot/RobotMap.java [hunk 7]
```diff
@@ -51,7 +51,7 @@
 
   /* Climber */
   public static final int climberMasterSRXID = 10;
-  public static final int climberSlaveSPXID = 12; // @todo not on robot yet
+  public static final int climberSlaveSPXID = 2; // @todo not on robot yet
```
Comment (roshanr10) [inline src/main/java/frc/team4909/robot/RobotMap.java]: remove the todo comment

File: src/main/java/frc/team4909/robot/Robot.java (hunk 4 above)
Comment (TechplexEngineer) [reply src/main/java/frc/team4909/robot/Robot.java]: I would argue that the button need not be hidden.

File: src/main/java/frc/team4909/robot/subsystems/climber/commands/BothLiftDown.java [hunk 8]
```diff
@@ -12,7 +13,11 @@ public BothLiftDown(){
...
+
+        Robot.climberSubsystem.setStiltsClimbSpeed(-RobotConstants.liftClimbSpeed);
```
Comment (roshanr10) [inline src/main/java/frc/team4909/robot/subsystems/climber/commands/BothLiftDown.java]: uh why the new negative sign? what in this changed ?

File: src/main/java/frc/team4909/robot/subsystems/climber/commands/SetStiltPosition.java [hunk 9]
```diff
@@ -36,8 +36,9 @@ protected void execute() {
         double driveStiltNegSpeed = Robot.driverGamepad.getThresholdAxis(BionicF310.LT)  //Stilt drive backwards
         * RobotConstants.climberDriveSpeedManual;
 
-        Robot.climberSubsystem.setStiltsDriveSpeed(RobotConstants.climberDriveSpeedAuto); // Stilts via climnber
-        
+        if(Robot.elevatorSubsystem.getPosition() > 0){
+            Robot.climberSubsystem.setStiltsDriveSpeed(RobotConstants.climberDriveSpeedAuto); // Stilts via climnber
```
Comment (roshanr10) [inline src/main/java/frc/team4909/robot/subsystems/climber/commands/SetStiltPosition.java]: the positive is also now up, so that may change the, just FYI

**Comment by roshanr10** [comment]
all the pending changes for Reading should be merged into here, tested and then merged to master

File: src/main/java/frc/team4909/robot/subsystems/climber/commands/BothLiftDown.java (hunk 1 above)
Comment (ashwinc12) [reply src/main/java/frc/team4909/robot/subsystems/climber/commands/BothLiftDown.java]: This was just for testing purpose to make sure the command was being called using the POV

File: src/main/java/frc/team4909/robot/subsystems/climber/commands/BothLiftUp.java (hunk 2 above)
Comment (ashwinc12) [reply src/main/java/frc/team4909/robot/subsystems/climber/commands/BothLiftUp.java]: Same

File: src/main/java/frc/team4909/robot/Robot.java (hunk 3 above)
Comment (ashwinc12) [reply src/main/java/frc/team4909/robot/Robot.java]: Stilts up and down only is controlled through the RIght Joystick of the climber gamepad.

File: src/main/java/frc/team4909/robot/subsystems/climber/commands/SetStiltPosition.java (hunk 9 above)
Comment (ashwinc12) [reply src/main/java/frc/team4909/robot/subsystems/climber/commands/SetStiltPosition.java]: Good point. I'll change later

**Comment by ashwinc12** [comment]
A full climb should be tested prior to Reading as well. Never went through a full cycle.

**Comment by roshanr10** [comment]
Full climb was tested several times after you left. We should be all set unless we want to retune the subsystems but YMMV
